src/dms/core/EdFi.DataManagementService.Core/ApiSchema/ApiSchemaDocument.cs

```diff
@@ -73,15 +73,54 @@ public ProjectNamespace GetMappedProjectName(string projectName)
     /// </summary>
     public List<JsonNode> GetAllProjectSchemaNodes()
     {
-        JsonNode schema = _apiSchemaRootNode;
+        JsonNode projectSchemasNode =
+            _apiSchemaRootNode["projectSchemas"]
+            ?? throw new InvalidOperationException("Expected ProjectSchemas node to exist.");
 
-        var projectSchemasNode = schema["projectSchemas"];
+        return projectSchemasNode.SelectNodesFromPropertyValues();
+    }
+
+    /// <summary>
+    /// Finds the CoreOpenApiSpecification, if this is a data standard ApiSchemaDocument. Returns null if not.
+    /// </summary>
+    public JsonNode? FindCoreOpenApiSpecification()
+    {
+        bool isExtensionProject = _apiSchemaRootNode.SelectRequiredNodeFromPathAs<bool>(
+            "$.projectSchemas['ed-fi'].isExtensionProject",
+            _logger
+        );
 
-        if (projectSchemasNode == null)
+        if (isExtensionProject)
         {
-            throw new InvalidOperationException("Expected ProjectSchmas node to exist.");
+            return null;
         }
 
-        return projectSchemasNode.SelectNodesFromPropertyValues();
+        return _apiSchemaRootNode.SelectRequiredNodeFromPath(
+            "$.projectSchemas['ed-fi'].coreOpenApiSpecification",
+            _logger
+        );
+    }
+
+    /// <summary>
+    /// Finds the OpenApiExtensionFragments, if this is an extension ApiSchemaDocument. Returns null if not.
+    /// </summary>
+    public JsonNode? FindOpenApiExtensionFragments()
+    {
+        // DMS-497 will fix: TPDM is hardcoded until we remove projectSchemas from ApiSchema.json - making one project per file
```

Comment:
Three places in code marked with DMS-497. The DMS-497 ticket references them as well.